src/dsl/validation.rs

```diff
@@ -398,4 +471,142 @@ mod tests {
         assert_eq!(validation_result.is_ok(), true);
         assert_eq!(deserialization_result.is_ok(), true);
     }
+
+    #[test]
+    fn validate_check_failure_message_expect_ok() {
+        let input = r#"
+            id: 156F64
+            name: Corosync configuration file
+            group: Corosync
+            description: |
+              Corosync `token` timeout is set to expected value
+            remediation: |
+              ## Abstract
+              The value of the Corosync `token` timeout is not set as recommended.
+              ## Remediation
+              ...
+            facts:
+              - name: corosync_token_timeout
+                gatherer: corosync.conf
+            values:
+              - name: expected_token_timeout
+                default: 5000
+                conditions:
+                  - value: 30000
+                    when: env.provider == "azure" || env.provider == "aws"
+                  - value: 20000
+                    when: env.provider == "gcp"
+            expectations:
+              - name: timeout
+                expect: facts.corosync_token_timeout == values.expected_token_timeout
+                failure_message: Expectation not met ${facts.carbonara}
```

Comment:
It is obvious you like carbonara hehe
I would put here `facts.corosync_token_timeout` to resemblance a real test, but well, not a big deal hehe